[bug] Test "change_branch_in_root_commit" fails if the git default branch is not master

### Environment details

* Operating System+version: macOS Ventura 13.2
* Compiler+version: Apple clang version 14.0.0 (clang-1400.0.29.202)
* Conan version: develop (1.58 as latest stable)
* Python version: 3.11.1


### Steps to reproduce

1. Check your default branch: `git config --global --get init.defaultBranch`
2. If it is "master" change it to something else: `git config --global init.defaultBranch main`
3. Run the tests (after the documented developer set up): `python -m tox -- conans/test/unittests/tools/scm/test_git_get_commit.py`

### Logs

The output for the tests:
```
========================================================================= test session starts ==========================================================================
platform darwin -- Python 3.11.1, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
cachedir: .tox/py/.pytest_cache
rootdir: /Users/gyula.gubacsi/Contrib/conan, configfile: pytest.ini
plugins: xdist-3.2.0
collected 4 items

conans/test/unittests/tools/scm/test_git_get_commit.py F...                                                                                                      [100%]

=============================================================================== FAILURES ===============================================================================
__________________________________________________________________ test_change_branch_in_root_commit ___________________________________________________________________

    def test_change_branch_in_root_commit():
        """
        https://github.com/conan-io/conan/issues/10971#issuecomment-1089316912
        """
        c = TestClient()
        conanfile = MockConanfile({})
        c.save({"root.txt": "", "subfolder/subfolder.txt": ""})
        c.run_command("git init .")
        c.run_command('git config user.name myname')
        c.run_command('git config user.email myname@mycompany.com')
        c.run_command("git add .")
        c.run_command('git commit -m "initial commit"')
        c.run_command("git checkout -b change_branch")
        c.save({"subfolder/subfolder.txt": "CHANGED"})
        c.run_command("git add .")
        c.run_command('git commit -m "second commit"')
>       c.run_command("git checkout master")

conans/test/unittests/tools/scm/test_git_get_commit.py:25:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
conans/test/utils/tools.py:582: in run_command
    self._handle_cli_result(command, assert_error=assert_error, error=ret)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <conans.test.utils.tools.TestClient object at 0x102762350>, command = 'git checkout master', assert_error = False, error = 1

    def _handle_cli_result(self, command, assert_error, error):
        if (assert_error and not error) or (not assert_error and error):
            if assert_error:
                msg = " Command succeeded (failure expected): "
            else:
                msg = " Command failed (unexpectedly): "
            exc_message = "\n{header}\n{cmd}\n{output_header}\n{output}\n{output_footer}\n".format(
                header='{:-^80}'.format(msg),
                output_header='{:-^80}'.format(" Output: "),
                output_footer='-' * 80,
                cmd=command,
                output=self.out
            )
>           raise Exception(exc_message)
E           Exception:
E           ------------------------ Command failed (unexpectedly): ------------------------
E           git checkout master
E           ----------------------------------- Output: ------------------------------------
E           error: pathspec 'master' did not match any file(s) known to git
E
E           --------------------------------------------------------------------------------

conans/test/utils/tools.py:598: Exception
```

--- a/conans/test/unittests/tools/scm/test_git_get_commit.py
+++ b/conans/test/unittests/tools/scm/test_git_get_commit.py
@@ -1,5 +1,4 @@
 import os
-import pytest
 
 from conan.tools.scm import Git
 from conans.tools import chdir
@@ -7,7 +6,6 @@ from conans.test.utils.mocks import MockConanfile
 from conans.test.utils.tools import TestClient
 
 
-@pytest.mark.xfail(reason="This test is failing if the environment has other than master as default branch", strict=True)
 def test_change_branch_in_root_commit():
     """
     https://github.com/conan-io/conan/issues/10971#issuecomment-1089316912
@@ -16,6 +14,7 @@ def test_change_branch_in_root_commit():
     conanfile = MockConanfile({})
     c.save({"root.txt": "", "subfolder/subfolder.txt": ""})
     c.run_command("git init .")
+    c.run_command("git checkout -B master")
     c.run_command('git config user.name myname')
     c.run_command('git config user.email myname@mycompany.com')
     c.run_command("git add .")

--- a/conans/test/unittests/tools/scm/test_git_get_commit.py
+++ b/conans/test/unittests/tools/scm/test_git_get_commit.py
@@ -1,4 +1,5 @@
 import os
+import pytest
 
 from conan.tools.scm import Git
 from conans.tools import chdir
@@ -6,6 +7,7 @@ from conans.test.utils.mocks import MockConanfile
 from conans.test.utils.tools import TestClient
 
 
+@pytest.mark.xfail(reason="This test is failing if the environment has other than master as default branch", strict=True)
 def test_change_branch_in_root_commit():
     """
     https://github.com/conan-io/conan/issues/10971#issuecomment-1089316912

